src/textual/message_pump.py

```diff
@@ -29,6 +32,32 @@ class MessagePumpClosed(Exception):
     pass
 
 
+T = TypeVar("T")
+
+
+def _is_coroutine_callable(call: Callable) -> bool:
```

Comment:
@willmcgugan I'm curious what your thoughts are on re-using these methods to allow users to define their `on_key`, etc. methods as sync or async, sorta how Starlette lets you define endpoints as sync or async and automatically runs sync endpoints in a threadpool? It would just be a couple lines in `_callback.py` I think.

Reply:
It would be good thought to only inspect the method once, then wrap it and store the wrapped version. This might require a mapping somewhere of `{user_method: possibly_wrapped_method}`.